sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -206,6 +216,64 @@ def test_partial_dependence_helpers(est, method, target_feature):
     assert np.allclose(pdp, mean_predictions, rtol=rtol)
 
 
+def test_recursion_decision_tree_vs_forest_and_gbdt():
+    # Make sure that the recursion method gives the same results on a
+    # DecisionTreeRegressor and a GradientBoostingRegressor or a
+    # RandomForestRegressor with 1 tree and equivalent parameters.
+
+    # Purely random dataset to avoid correlated features
+    n_samples = 100
+    n_features = 5
+    X = np.random.RandomState(0).randn(n_samples, n_features)
```

Comment:
```suggestion
    seed = 0
    rng = np.random.RandomState(seed)
    X, y = rng.randn(n_samples, n_features), rng.randn(n_samples)
```

Reply:
Just curious, why do you want to declare `seed` and use it here and not below for `y` or for `random_state=0`?

Reply:
Um actually I want to declare rng and use it everywhere. Is there a reason to repeat np.random.RandomState?

Reply:
I can't remember, using a rng now.

It made me realize the test isn't as stable as I thought (about 7 failure over 100 runs). The failures come from the tree vs gb comparisons only (tree vs forest is always passing). This is a bit strange but I think the test is still OK

Reply:
do you think that making it with iris, for instance, would be better since there is a pattern?

Reply:
iris would work but there are only 3 targets with iris so that's not a super interesting problem